### PR TITLE
Retry operator issue

### PR DESCRIPTION
**Pull request by davidwdan**
Check to see if we have a disposable before trying to dispose

### CODE REVIEW

File: lib/Rx/Operator/RetryOperator.php [hunk 1]
```diff
@@ -41,7 +41,9 @@ function ($error) use ($observable, $observer, &$disposable, &$getNewObserver, $
                         $observer->onError($error);
                         return;
                     }
-                    $disposable->dispose();
+                    if ($disposable) {
```
Comment (asm89) [inline lib/Rx/Operator/RetryOperator.php]: When does this become `null`? `$getNewObserver` is not called until line 56.

Comment (davidwdan) [inline lib/Rx/Operator/RetryOperator.php]: Sorry, I missed this comment.  I'll change it to a SerializeDisposable, so it's clear what's going on.

**Comment by asm89** [comment]
@davidwdan Can you add a test that exposes where it previously went wrong?

**Comment by davidwdan** [comment]
@asm89 added a test

File: lib/Rx/Operator/RetryOperator.php [hunk 2]
```diff
@@ -31,18 +32,23 @@ public function __invoke(
         ObservableInterface $observable,
         ObserverInterface $observer,
         SchedulerInterface $scheduler = null
-    ) {
-        $getNewObserver = function () use ($observable, $observer, &$disposable, &$getNewObserver, $scheduler) {
+    )
+    {
```
Comment (asm89) [inline lib/Rx/Operator/RetryOperator.php:32]: these should be on the same line

File: lib/Rx/Operator/RetryOperator.php [hunk 3]
```diff
@@ -31,18 +32,23 @@ public function __invoke(
...
+
```
Comment (asm89) [inline lib/Rx/Operator/RetryOperator.php]: nit: whitespace

**Pull request merged**